Mathlib/SetTheory/Surreal/Division.lean

```diff
@@ -0,0 +1,714 @@
+            simp only [Sum.elim_inr, quot_sub, quot_add, quot_zero_mul, add_sub_cancel_left]
+            have := onag_1_10_i_left j
+            rw [pos_num_eq_normalization x_num x_pos] at this
+            exact this
+      · case _ val =>
+          rcases val with ⟨i, j⟩
+          specialize this (Sum.inr ⟨i, j⟩)
+          rw [pos_num_eq_normalization x_num x_pos] at this
+          exact this
+
+    have onag_1_10_iii_right : ∀ j, 1 < ((normalization x) * x.inv').moveRight j := by
+      have := onag_1_10_iii_right' h3 h4 x_num inv_l inv_r (by rwa [x_rfl] at onag_1_10_ii) x_pos
+      rw [x_rfl]
+      intro i
+      cases i
+      · case _ val =>
+          rcases val with ⟨i | val, j⟩
+          · specialize this (Sum.inl ⟨i, j⟩)
+            rw [pos_num_eq_normalization x_num x_pos] at this
+            exact this
+          · cases val  -- the case `x' = 0`, not mentioned in ONAG
+            rw [Game.PGame.lt_iff_game_lt]
+            simp only [normalization, insertLeft, inv', mk_mul_moveRight_inl]
+            simp only [Sum.elim_inr, quot_sub, quot_add, quot_zero_mul, add_sub_cancel_left]
+            have := onag_1_10_i_right j
+            rw [pos_num_eq_normalization x_num x_pos] at this
+            exact this
+      · case _ val =>
+          rcases val with ⟨i, j⟩
+          specialize this (Sum.inr ⟨i, j⟩)
+          rw [pos_num_eq_normalization x_num x_pos] at this
+          exact this
+
+    have onag_1_10_iv' :
+        mk (normalization x) (num_of_normalization_num x_num x_pos) *
+          mk (x.inv') onag_1_10_ii = 1 := by
+      simp only [← mk_mul, one_def, mk_eq_mk]
+      rw [equiv_def]
+      constructor <;> rw [le_iff_forall_lf]
+      · simp only [one_rightMoves, IsEmpty.forall_iff, and_true]
+        intro i
+        have := onag_1_10_iii_left i
+        exact lf_of_lt this
+      · simp only [one_leftMoves, one_moveLeft, forall_const]
+        constructor
+        · rw [lf_iff_exists_le]
+          left
+          rcases x with ⟨xl, xr, xL, xR⟩
+          use toLeftMovesMul (Sum.inl (Sum.inr (), InvTy.zero))
+          simp only [mul_moveLeft_inl, inv', moveLeft_mk, invVal]
+          rw [Game.PGame.le_iff_game_le]
+          simp only [quot_sub, quot_add, quot_mul_zero, add_sub_cancel_right]
+          simp only [normalization, insertLeft, moveLeft_mk, Sum.elim_inr, quot_zero_mul, le_refl]
+        · intro j
+          have := onag_1_10_iii_right j
+          exact lf_of_lt this
+
+    have onag_1_10_iv : x * x.inv' ≈ 1 := by
+      suffices mk x x_num * mk x.inv' onag_1_10_ii = 1 by
+        simp only [← mk_mul, one_def, mk_eq_mk] at this
+        exact this
+      rwa [pos_num_eq_normalization x_num x_pos]
+
+    exact ⟨onag_1_10_ii, onag_1_10_iv⟩
+
+lemma onag_1_10_ii : (x.inv').Numeric := (onag_1_10 x_num x_pos).1
+
+lemma onag_1_10_iv : x * x.inv' ≈ 1 := (onag_1_10 x_num x_pos).2
+
+end Positive
+
+lemma inv_surreal (hx : ¬ x ≈ 0) : x * x⁻¹ ≈ 1 := by
```

Comment:
```suggestion
protected lemma mul_inv_self (hx : ¬ x ≈ 0) : x * x⁻¹ ≈ 1 := by
```

Reply:
With `protected`, this doesn't work - then Lean cannot find this `mul_inv_self` when it is used later in this file. I removed the `protected` and it works.

Reply:
The `protected` keyword makes it so that you have to specify the namespace explicitly, i.e. `Surreal.mul_inv_self`. That way, the name of this lemma can correspond to the analogous [`mul_inv_self`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Group/Defs.html#mul_inv_self) in the root namespace without clashing with it.

Reply:
According to [here](https://github.com/leanprover-community/mathlib4/blob/74ed7693458847faae657f3489f6d1d228607280/Mathlib/Algebra/Group/Defs.lean#L1073) `mul_inv_self` is deprecated in favor of `mul_inv_cancel` so perhaps I should use `mul_inv_cancel` here also ?

Reply:
I changed `mul_inv_self` to `mul_inv_cancel`, added `protected` keyword, and also renamed `mul_inv'_self` to `mul_inv'_cancel`.